pkg/template/nstemplatetier.go

```diff
@@ -0,0 +1,47 @@
+package template
+
+import (
+	"context"
+	"fmt"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// NSTemplates the templates along with their revision number for a given tier
+type NSTemplates map[string]RevisionedTemplate
+
+// RevisionedTemplate a template along with its revision number
+type RevisionedTemplate struct {
+	Revision string
+	Template string
+}
+
+// GetNSTemplates gets the templates configured in the NSTemplateTier resource
+// which is fetched from the host cluster.
+func GetNSTemplates(tierName string) (NSTemplates, error) {
+	// retrieve the address of the host cluster in the
+	host, ok := cluster.GetHostCluster()
+	if !ok {
+		return nil, fmt.Errorf("unable to connect to the Host cluster: unknown cluster")
+	}
```

Comment:
I think that you should also verify that the cluster is ready - see here: 
https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccountstatus/useraccount_status_controller.go#L101-L106

Reply:
thanks for the pointer! I added that check in edc3e2f

Reply:
removed in https://github.com/codeready-toolchain/member-operator/commit/21bb803d4a1d43875aabaf0f26d516f94a817ad2 after our fruitful discussion ;)

Reply:
(responded to wrong comment)